### PR TITLE
Return error if Get() fails in Prefetching Filter blocks

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: Right now all I/O failures under
PartitionFilterBlock::CacheDependencies() is swallowed. Return error in
case prefetch fails.

Test Plan: make check -j64

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by akankshamahajan15** [comment]
Failure:  ci/circleci: build-linux-clang10-ubsan  not related to this change.

File: file/file_prefetch_buffer.cc [hunk 1]
```diff
@@ -91,17 +91,11 @@ Status FilePrefetchBuffer::Prefetch(const IOOptions& opts,
...
-  if (!s.ok() || result.size() < read_len) {
-    // Fake an IO error to force db_stress fault injection to ignore
-    // truncated read errors
-    IGNORE_STATUS_IF_ERROR(Status::IOError());
-  }
-#endif
-  if (s.ok()) {
-    buffer_offset_ = rounddown_offset;
-    buffer_.Size(static_cast<size_t>(chunk_len) + result.size());
+  if (!s.ok()) {
```
Comment (anand1976) [inline file/file_prefetch_buffer.cc:94]: We still need to call ```IGNORE_STATUS_IF_ERROR()``` if ```result.size() < read_len```. That's not really an error and can happen if the readahead goes past the end of file.

File: table/block_based/partitioned_filter_block.cc [hunk 2]
```diff
@@ -477,17 +475,18 @@ void PartitionedFilterBlockReader::CacheDependencies(const ReadOptions& ro,
         prefetch_buffer.get(), ro, handle, UncompressionDict::GetEmptyDict(),
         &block, BlockType::kFilter, nullptr /* get_context */, &lookup_context,
         nullptr /* contents */);
-
-    assert(s.ok() || block.GetValue() == nullptr);
```
Comment (anand1976) [inline table/block_based/partitioned_filter_block.cc:481]: I think this assertion is still valid?

File: table/block_based/partitioned_filter_block.cc [hunk 3]
```diff
@@ -426,12 +426,7 @@ void PartitionedFilterBlockReader::CacheDependencies(const ReadOptions& ro,
   Status s = GetOrReadFilterBlock(false /* no_io */, nullptr /* get_context */,
                                   &lookup_context, &filter_block);
   if (!s.ok()) {
-    ROCKS_LOG_WARN(rep->ioptions.info_log,
```
Comment (anand1976) [inline table/block_based/partitioned_filter_block.cc:429]: We should retain this log message

File: table/block_based/partitioned_filter_block.cc (hunk 2 above)
Comment (akankshamahajan15) [reply table/block_based/partitioned_filter_block.cc:481]: yes it is. I recently got to know that assertions are excluded in optimized builds.

**Review by anand1976 — APPROVED** [review]

**Pull request closed without merging**